packages/boot/tools/supports.ts

```diff
@@ -150,30 +154,27 @@ export const makeProposalExtractor = ({ childProcess, fs }: Powers) => {
   };
 
   const buildAndExtract = async (builderPath: string) => {
-    const scriptEnv = Object.assign(Object.create(process.env));
-
-    const pkgPath = getPkgPath('builders');
-    const tmpDir = await fsAmbientPromises.mkdtemp(join(pkgPath, 'proposal-'));
-
-    const scriptPath = await importSpec(builderPath);
+    const tmpDir = await fsAmbientPromises.mkdtemp(
+      join(getPkgPath('builders'), 'proposal-'),
+    );
 
-    // XXX use '@agoric/inter-protocol'?
-    const out = await runPackageScript(tmpDir, scriptPath, scriptEnv);
-    const built = parseProposalParts(out.toString());
+    const built = parseProposalParts(
+      runPackageScript(
+        tmpDir,
+        await importSpec(builderPath),
+        process.env,
+      ).toString(),
+    );
 
-    const loadAndRmPkgFile = async fileName => {
-      const filePath = join(tmpDir, fileName);
-      const content = await fs.readFile(filePath, 'utf8');
-      await fs.rm(filePath);
-      return content;
-    };
+    const loadPkgFile = fileName => fs.readFile(join(tmpDir, fileName), 'utf8');
 
     const evalsP = Promise.all(
       built.evals.map(async ({ permit, script }) => {
         const [permits, code] = await Promise.all([
-          loadAndRmPkgFile(permit),
-          loadAndRmPkgFile(script),
+          loadPkgFile(permit),
+          loadPkgFile(script),
         ]);
+        void fsAmbientPromises.rm(tmpDir, { recursive: true, force: true });
```

Comment:
This is the only line I find a little suspect.  Does `void`ing instead of `await`ing this promise mean the script might exit before `tmpDir` is fully removed?

Reply:
Because it's not awaited, if the Node process reaches termination before it's done I suppose it would be aborted. That seems very unlikely but on the outside chance it's not worth slowing down the test on that file IO.

Is a comment to that effect sufficient or do you think it's worth awaiting?

Reply:
I'll go with the comment since that was the request